### ISSUE TITLE
DMS 0.9.7 Insert trigger fails with composite primary key

### ISSUE THREAD

**Issue by VagueGit** [Mimetis/Dotmim.Sync#886]
Trying to sync a table with a composite primary key on SQL Server via NetCore API to SQLite.
On the sql server we have
```
CREATE TABLE [dbo].[LaborItems](
	[TaskName] [varchar](15) NOT NULL,
	[MaterialCost] [decimal](9, 2) NOT NULL,
	[MinutesSmall] [smallint] NOT NULL,
	[MinutesLarge] [smallint] NOT NULL,
 CONSTRAINT [PK_LaborItems] PRIMARY KEY CLUSTERED 
(
	[TaskName] ASC,
	[MaterialCost] ASC
)
```
The SQLite table is created by the app (not DMS)
```
CREATE TABLE LaborItems (
TaskName TEXT NOT NULL COLLATE NOCASE CHECK (length(TaskName) <= 15), 
MaterialCost NUMERIC NOT NULL CHECK (MaterialCost BETWEEN -9999998.99 AND 9999999.99), 
MinutesSmall INTEGER NOT NULL CHECK (MinutesSmall BETWEEN 0 AND 32767), 
MinutesLarge INTEGER NOT NULL CHECK (MinutesLarge BETWEEN 0 AND 32767), 
PRIMARY KEY(TaskName, MaterialCost));
```
DMS provisions both the server and the client dbs
```
CREATE TABLE [LaborItems_tracking] (
[TaskName] text NOT NULL COLLATE NOCASE, 
[MaterialCost] numeric NOT NULL COLLATE NOCASE, 
[update_scope_id] [text] NULL COLLATE NOCASE, 
[timestamp] [integer] NULL, 
[sync_row_is_tombstone] [integer] NOT NULL default(0), 
[last_change_datetime] [datetime] NULL, 
 PRIMARY KEY ([TaskName], [MaterialCost]));

CREATE INDEX [LaborItems_tracking_timestamp_index] ON [LaborItems_tracking] (
	 [timestamp] ASC
	,[update_scope_id] ASC
	,[sync_row_is_tombstone] ASC
	,[TaskName] ASC
	,[MaterialCost] ASC
);

CREATE TRIGGER [LaborItems_insert_trigger] AFTER INSERT ON [LaborItems] 

BEGIN
-- If row was deleted before, it already exists, so just make an update
	INSERT OR REPLACE INTO [LaborItems_tracking] (
		[TaskName]
		,[update_scope_id]
		,[timestamp]
		,[sync_row_is_tombstone]
		,[last_change_datetime]
	) 
	VALUES (
		new.[TaskName]
		,NULL
		,replace(strftime('%Y%m%d%H%M%f', 'now'), '.', '')
		,0
		,datetime('now')
	);
END;
```
The exception thrown on insert is
SQLite Error 19: 'NOT NULL constraint failed: LaborItems_tracking.MaterialCost

**Comment by Mimetis** [comment]
I followed your script, creating the server database, then creating the client database with an existing table
Then I tried a sync, successfully

The trigger generated on the fly:

``` sql
CREATE TRIGGER [LaborItems_insert_trigger] AFTER INSERT ON [LaborItems] 

BEGIN
-- If row was deleted before, it already exists, so just make an update
	INSERT OR REPLACE INTO [LaborItems_tracking] (
		[TaskName]
		,[MaterialCost]
		,[update_scope_id]
		,[timestamp]
		,[sync_row_is_tombstone]
		,[last_change_datetime]
	) 
	VALUES (
		new.[TaskName]
		,new.[MaterialCost]
		,NULL
		,replace(strftime('%Y%m%d%H%M%f', 'now'), '.', '')
		,0
		,datetime('now')
	);
END
```

Any chance you made some changes on the table (playing with primary keys) after a first sync and you forgot to delete the triggers ?

**Comment by VagueGit** [comment]
Hmm ... stepping through ... the trigger is created correctly as you say, then is later recreated incorrectly. Will investigate and report back

**Comment by VagueGit** [comment]
Is this a bug in `localOrchestrator.GetSchemaAsync(syncSetup);` that only returns one column of a composite primary key?

**Comment by Mimetis** [comment]
GOOD CATCH !!

You spotted a bug in the **SQLite** provider @VagueGit :)

The sql statement is incorrect:
``` cs
string commandColumn = $"SELECT * FROM pragma_table_info('{unquotedTableName}') where pk = 1;";
```

And the good statement should be:
``` cs
string commandColumn = $"SELECT * FROM pragma_table_info('{unquotedTableName}') where pk <> 0 order by pk asc;";
```

Thanks @VagueGit for spotted this one.
Will be fixed in the next `0.9.8 beta` version.

**Comment by VagueGit** [comment]
Confirmed this is fixed in DMS 0.9.8-beta-0994. A very welcome xmas present thank you.
Joyeux Noël
